### DISCUSSION TITLE
Desktop Cruiser GL

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#523] [Q&A] [answered]
Because there are persistent issues with OpenGL on desktop (Windows, macOS, Linux),
e.g. no help with [Apple Arm](https://github.com/mapsforge/vtm/issues/1004), issues on Linux, etc.

I'm thinking of stopping the OpenGL version of Cruiser on desktop
and continue with the Java (Mapsforge) version.

_Note: the discussion is about the desktop (not Android)!_

**Comment by Leeb64** [comment]
I don't use it anyway.
It is better:https://brouter.de/essbee/
It is very easy to add your own BRouter profiles there.
-> 1:1 route planning!

**Reply by 0709wiwiwi** [reply]
In terms of presentation, the desktop version is top. 
Ideal for a demo to people who have not yet discovered Cruiser.
https://youtu.be/U7HoC3JGNWA

**Comment by devemux86** [comment, marked as answer]
The only major difference are the map themes (like Biker or Motorider) that do not work on regular desktop Cruiser.
Otherwise both map engines offer the same things, except map rotation / tilt which is not important on the desktop.

**Comment by OAStubby** [comment]
Is it possible to implement 3D buildings in the Mapsforge version as well?

**Reply by devemux86** [reply]
This is not possible because 3D buildings require OpenGL.

**Comment by devemux86** [comment]
If Mapsforge had or will get map rotation and continuous zoom in the future,
then we do not need OpenGL on Android either. :slightly_smiling_face:

**Reply by Klaus-Tockloth** [reply]
Mapsforge map rotation: For me one the most important improvements. Thanks for thinking about it.

**Reply by devemux86** [reply]
@Klaus-Tockloth 

It would be nice, but unfortunately there is not much help from other developers
to implement new features into the Mapsforge engine.

For example, Locus uses Mapsforge maps, but has implemented its own map engine with map rotation.

**Comment by MotoUKRider** [comment]
Mixed feelings - I find it useful for easy offline mapping when I want to look at places, but less so now that I've mastered the art of Garmin maps, mkgmap/osmium and the CPU waiting game.

It helped a LOT with the original theme changes and I suppose people like either an App or a Browser experience (plus all the cloud stuff) and may be happy to pay a monthly fee.  Personally, I like the offline experience - I am more than happy to blame me if I haven't sync'd the phone with the computer...

**Comment by ramondevic** [comment]
I occasionally use Cruiser Desktop, I like it but it seems very neglected, especially in terms of favorites.

To prepare the routes I use Kurviger which is fabulous.

But apparently the development of Cruiser is focusing on the preparation of the routes from the mobile?
:(

**Reply by devemux86** [reply]
> But apparently the development of Cruiser is focusing on the preparation of the routes from the mobile?

Offline routing with BRouter is currently only available on mobile.

Or can also use the [BRouter website](https://brouter.de/brouter-web/):
- https://github.com/devemux86/cruiser/discussions/272

**Reply by ramondevic** [reply]
Ok, I'll try  [BRouter website](https://brouter.de/brouter-web/)

Thanks.
